src/values/SymbolValue.js

```diff
@@ -9,11 +9,11 @@
 
 /* @flow */
 
-import { PrimitiveValue } from "./index.js";
+import { PrimitiveValue, AbstractValue } from "./index.js";
 import type { Realm } from "../realm.js";
 
 export default class SymbolValue extends PrimitiveValue {
-  constructor(realm: Realm, desc?: string, intrinsicName?: string) {
+  constructor(realm: Realm, desc?: string | AbstractValue, intrinsicName?: string) {
```

Comment:
The code might become simpler if the desc/$Description in SymbolValue is always a Value (for which holds that `!{value}.mightNotBeString`.